File: python/News-Article-Scraper/Python/extract_article_links.py
from bs4 import BeautifulSoup
import requests


def parse_sitemap() -> list:
    response = requests.get("https://www.example.com/sitemap.xml")
    if response.status_code != 200:
        return None
    xml_as_str = response.text

    soup = BeautifulSoup(xml_as_str, "lxml")
    loc_elements = soup.find_all("loc")
    links = []
    for loc in loc_elements:
        links.append(loc.text)

    print(f'Found {len(links)} links')
    return links


if __name__ == '__main__':
    links = parse_sitemap()

File: python/News-Article-Scraper/JavaScript/extract_article_links.js
const cheerio = require("cheerio");
const axios = require("axios");
url = `https://www.patrika.com/googlenewssitemap1.xml`;
let links = [];
async function getLinks() {
    try {
        const response = await axios.get(url);
        const $ = cheerio.load(response.data, { xmlMode: true });
        all_loc = $('loc')
        all_loc.each(function () {
            links.push($(this).text())
        })
        console.log(links.length + ' links found.')

    } catch (error) {
        console.error(error);
    }
}
getLinks();
